homeassistant/components/splunk/__init__.py

```diff
@@ -102,22 +113,23 @@ async def splunk_event_listener(event):
 
         payload = {
             "time": event.time_fired.timestamp(),
-            "host": name,
             "event": {
                 "domain": state.domain,
                 "entity_id": state.object_id,
                 "attributes": dict(state.attributes),
                 "value": _state,
             },
         }
+        if name:
+            payload["host"] = name
 
         try:
-            await event_collector.queue(json.dumps(payload, cls=JSONEncoder), send=True)
+            await splunk.queue(json.dumps(payload, cls=JSONEncoder), send=True)
         except SplunkPayloadError as err:
-            if err.status == HTTPStatus.UNAUTHORIZED:
-                _LOGGER.error(err)
-            else:
-                _LOGGER.warning(err)
+            if err.code in [1, 2, 3, 4]:
+                # Auth failure
+                raise ConfigEntryAuthFailed() from err
+            _LOGGER.warning(err)
         except ClientConnectionError as err:
             _LOGGER.warning(err)
```

Comment:
How are events batched when sent? How often will the client send the data from the queue?

Reply:
This is a pre-existing problem. I'll quickly code up a fix to raise once as a seperate PR.